integrationTests/multiShard/block/executingMiniblocksSc_test.go

```diff
@@ -235,6 +236,109 @@ func TestProcessWithScTxsJoinAndRewardTwoNodesInShard(t *testing.T) {
 	checkRootHashes(t, nodes, idxProposers)
 }
 
+// TestShouldProcessWithScTxsJoinNoCommitShouldProcessedByValidators tests the following scenario:
+// There are 2 shard and 1 meta, each with one proposer and one validator.
+// Shard 1's proposer deploys a SC. There is 1 round for proposing block that will create the SC account.
+// Shard 0's proposer sends a joinGame SC call tx, proposes a block (not committing it) and the validator
+// should be able to sync it.
+func TestShouldProcessWithScTxsJoinNoCommitShouldProcessedByValidators(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	scCode, err := ioutil.ReadFile(agarioFile)
+	assert.Nil(t, err)
+
+	maxShards := uint32(2)
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+	advertiserAddr := integrationTests.GetConnectableAddress(advertiser)
+
+	nodeProposerShard0 := integrationTests.NewTestProcessorNode(maxShards, 0, 0, advertiserAddr)
+	nodeValidatorShard0 := integrationTests.NewTestProcessorNode(maxShards, 0, 0, advertiserAddr)
+
+	nodeProposerShard1 := integrationTests.NewTestProcessorNode(maxShards, 1, 1, advertiserAddr)
+	hardCodedSk, _ := hex.DecodeString("5561d28b0d89fa425bbbf9e49a018b5d1e4a462c03d2efce60faf9ddece2af06")
+	hardCodedScResultingAddress, _ := hex.DecodeString("000000000000000000005fed9c659422cd8429ce92f8973bba2a9fb51e0eb3a1")
+	nodeProposerShard1.LoadTxSignSkBytes(hardCodedSk)
+	nodeValidatorShard1 := integrationTests.NewTestProcessorNode(maxShards, 1, 1, advertiserAddr)
+
+	nodeProposerMeta := integrationTests.NewTestProcessorNode(maxShards, sharding.MetachainShardId, 0, advertiserAddr)
+	nodeValidatorMeta := integrationTests.NewTestProcessorNode(maxShards, sharding.MetachainShardId, 0, advertiserAddr)
+
+	nodes := []*integrationTests.TestProcessorNode{
+		nodeProposerShard0,
+		nodeProposerShard1,
+		nodeProposerMeta,
+		nodeValidatorShard0,
+		nodeValidatorShard1,
+		nodeValidatorMeta,
+	}
+
+	idxProposerShard0 := 0
+	idxProposerShard1 := 1
+	idxProposerMeta := 2
+	idxProposers := []int{idxProposerShard0, idxProposerShard1, idxProposerMeta}
+	idxProposersWithoutShard1 := []int{idxProposerShard0, idxProposerMeta}
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Messenger.Close()
+		}
+	}()
+
+	for _, n := range nodes {
+		_ = n.Messenger.Bootstrap()
+	}
+
+	fmt.Println("Delaying for nodes p2p bootstrap...")
+	time.Sleep(stepDelay)
+
+	round := uint32(0)
+	round = incrementAndPrintRound(round)
+
+	initialVal := big.NewInt(10000000)
+	topUpValue := big.NewInt(500)
+	integrationTests.MintAllNodes(nodes, initialVal)
+
+	deployScTx(nodes, idxProposerShard1, string(scCode))
+	proposeBlockWithScTxs(nodes, round, idxProposers)
+	syncBlock(t, nodes, idxProposers, round)
+	round = incrementAndPrintRound(round)
+
+	nodeJoinsGame(nodes, idxProposerShard0, topUpValue, hardCodedScResultingAddress)
+	for i := 0; i < 10; i++ {
```

Comment:
Why 10? Is just a big enough number to ensure that the block with sc results would be processed?

Reply:
yes, added variable that explains what 10 is.